### ISSUE TITLE
Allow removing requirement for `TypeInfo` bounds on type params

### ISSUE THREAD

**Issue by ascjones** [paritytech/scale-info#95]
Polkadot has the type [`Xcm<Call>`](https://github.com/paritytech/polkadot/blob/5e3cd0bd801409443823ead7697605095f792c68/xcm/src/v0/mod.rs#L92) which has an `Order<Call>` and an `Order<()>` which in turn has an `Xcm<Call>`. We get a `overflow evaluating the requirement Vec<order::Order<()>>: TypeInfo`

Since `Call` is not directly encoded, see [`double_encoded`](https://github.com/paritytech/polkadot/blob/12a3e03fd1562d87fa498d3b63e697e0d32e1481/xcm/src/double_encoded.rs#L23), this type parameter does not need to implement `Encode/Decode/TypeInfo`

To get around this for other derives they are just removing the bounds:

```
#[derivative(Clone(bound = ""), Eq(bound = ""), PartialEq(bound = ""), Debug(bound = ""))]
#[codec(encode_bound())]
#[codec(decode_bound())]
```

On the surface we should be able to add `#[scale_info(bounds()))]` from https://github.com/paritytech/scale-info/pull/88 to erase the bounds. Though that is not enough on its own because there is still `TypeInfo` requirement for adding to [`type_params`](https://github.com/paritytech/scale-info/blob/bd6dd5dd21a90a3fabdc6ad7b90779a4b0e4d409/derive/src/lib.rs#L113).

So what we need to be able to do is control which type params are included in the `type_params` list, in addition to (or even instead of) the custom bounds.

**Comment by ascjones** [comment]
> Polkadot has the type Xcm<Call> which has an Order<Call> and an Order<()> which in turn has an Xcm<Call>. We get a `overflow evaluating the requirement Vec<order::Order<()>>: TypeInfo`

As suggested by the error message the issue is in caused by the auto added `Vec<order::Order<()>>: TypeInfo` bound. 

The reason we have such an apparently unnecessary bound is because there is not currently a good solution to differentiate between a generic type constructor e.g. `Vec<T>` and an alias to an associated type e.g. `BalanceOf<T>`. And we need the bound for the latter, so we just add both.

I have added a test to #88 which fixes this with a custom bound override: https://github.com/paritytech/scale-info/pull/88/commits/3a987993add90c83d783ee75cfe89d1faca16877

**Comment by ascjones** [comment]
Closed by #96
